### DISCUSSION TITLE
Get whole day Heart Rate data

### DISCUSSION THREAD

**Discussion by psychon7** [matin/garth#10] [Q&A] [answered]
In Garmin connect dailies endpoint in Health API 

we get timeOffsetHeartRateSamples": "{15: 75, 30: 75, 3180: 76, 3195: 65, 3210: 65, 3225: 73, 3240: 74, 3255: 74}", can I use this library to call the dailies endpoint? And should I have to use a garmin developer account consumer and secret key to use this ?

**Comment by matin** [comment, marked as answer]
> can I use this library to call the dailies endpoint?

~If you know the endpoint, you can call `garth.client.connectapi(path)`. Otherwise, if you create an issue, I can add it.~

Garth does not currently support the Garmin Health API

> And should I have to use a garmin developer account consumer and secret key to use this ?

~Not at all.~

Garth uses the same consumer key and secret as the Connect app, but those keys don't appear to work for the Health API.

**Reply by psychon7** [reply]
Thanks for the quick response 🩵.  So if im using https://apis.garmin.com/wellness-api/rest/dailies, I can simply call garth.client.connectapi(/wellness-api/rest/dailies) ?

**Reply by matin** [reply]
Try the following instead:

`garth.connectapi("/mobile-gateway/heartRate/forDate/2023-03-28")`

**Reply by psychon7** [reply]
This method returns heart rate every 2mins, but the Garmin health API https://apis.garmin.com/wellness-api/rest/dailies sample response in docs has heart rate every 15 seconds. I would like to know which method I should use to get. whole day Heart rate at 15-second interval the same as "timeOffsetHeartRateSamples"? 

[
  {
    "userId": "xxxxxxxxxxxxxxxxx",
    "summaryId": "x153a9f3-5a9478d4-6",
    "calendarDate": "2022-01-11",
    "activityType": "WALKING",
    "activeKilocalories": 321,
    "bmrKilocalories": 1731,
    "steps": 4210,
    "pushes": 3088,
    "distanceInMeters": 3146.5,
    "pushDistanceInMeters": 2095.2,
    "durationInSeconds": 86400,
    "activeTimeInSeconds": 12240,
    "startTimeInSeconds": 1452470400,
    "startTimeOffsetInSeconds": 3600,
    "moderateIntensityDurationInSeconds": 81870,
    "vigorousIntensityDurationInSeconds": 4530,
    "floorsClimbed": 8,
    "minHeartRateInBeatsPerMinute": 59,
    "maxHeartRateInBeatsPerMinute": 112,
    "averageHeartRateInBeatsPerMinute": 64,
    "restingHeartRateInBeatsPerMinute": 64,
    "timeOffsetHeartRateSamples": "{15: 75, 30: 75, 3180: 76, 3195: 65, 3210: 65, 3225: 73, 3240: 74, 3255: 74}",
    "source": "string",
    "stepsGoal": 4500,
    "pushesGoal": 3100,
    "intensityDurationGoalInSeconds": 1500,
    "floorsClimbedGoal": 18,
    "averageStressLevel": 43,
    "maxStressLevel": 87,
    "stressDurationInSeconds": 13620,
    "restStressDurationInSeconds": 7600,
    "activityStressDurationInSeconds": 3450,
    "lowStressDurationInSeconds": 6700,
    "mediumStressDurationInSeconds": 4350,
    "highStressDurationInSeconds": 108000,
    "stressQualifier": "stressful_awake"
  }
]

**Reply by matin** [reply]
Could you share the link to the page in the documentation that shows the URL and response you shared?

**Reply by psychon7** [reply]
https://healthapi.garmin.com/tools/apiDocs#/ - you need a Garmin developer account to access this page. One doubt of mine is that, can we only have access to the "timeOffsetHeartRateSamples" if we have developer account credentials or if the same data is also available on the public access route in Garth.

**Reply by matin** [reply]
I don't have a Garmin developer account, so I can't see that page.

Garth is only authenticating for the Connect API—as it's modeled around the Garmin Connect app.

Do any of the Garmin mobile apps or webpages show this information?

**Reply by matin** [reply]
They're different APIs

> The Garmin Connect APIs enable cloud-to-cloud based integrations. For real-time data and a direct connection between your mobile app and Garmin wearables, check out what is possible using our [Garmin Health SDKs](https://developer.garmin.com/health-sdk/overview/).

**Reply by psychon7** [reply]
I have a streamlit web app were i visualize the frequency domain features of the heart rate so that's why I require 15s granular samples. Which I can get in the Garmin endpoint using developer credentials. Would be great to get the same granular HR in the Python client too.. even if I have to use the developer credentials that would be ok..

**Reply by matin** [reply]
I can understand why you need granular data. Integrating the Health API is _currently_ outside the scope of Garth.

Out of curiosity, what insights are looking to gain from the frequency domain features of the HR? What's the goal?

I started Garth as a way to study correlation between stress, sleep, activity, etc. In particular, I've been trying to analyze ways to improve sleep. I use a combination of Notebooks and chatgpt's data analyzer to understand my data in a level of detail that would be impossible through the app or website.

**Reply by psychon7** [reply]
I'm looking into the frequency power spectrum to assess the autonomic nervous system by finding (LF/HF Ratio).  Trying to access the dominance between sympathetic and parasympathetic systems.
